app/javascript/components/App.jsx

```diff
@@ -1,22 +1,58 @@
 import React, { Component } from 'react';
 import Home from './Home';
+import SignUp from './SignUp';
+import SignIn from './SignIn';
+import SignOut from './SignOut';
 import {HashRouter as Router, Route, NavLink, Switch} from 'react-router-dom'
+import history from "../packs/history";
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state ={};
+  }
 
-class App extends Component {
   render() {
+    let registerNavigation;
```

Comment:
I don't think it is a good naming for variable. It takes as value either <RegisterNavigation /> (same as variable) or <SignOUtNavigation />. To my opinion navbarComponent looks better